File: 0456-132-pattern/0456-132-pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        int n=nums.size();
        int s3 = INT_MIN;
        stack<int> st;
        for(int i=n-1;i>=0;i--){
            if(nums[i]<s3) return true;
            else
                while(!st.empty() and nums[i]>st.top()){
                    s3 = st.top();
                    st.pop();
                }
                st.push(nums[i]);
        }
        return false;
    }
};

File: 0456-132-pattern/0456-132-pattern.rs
impl Solution {
    pub fn find132pattern(nums: Vec<i32>) -> bool {
    // let n = nums.len();
    let mut s3 = i32::MIN;
    let mut stack: Vec<i32> = Vec::new();
    for &num in nums.iter().rev(){
        if num<s3 {return true;}
        else
        {while !stack.is_empty() && num>*stack.last().unwrap(){
            s3 = stack.pop().unwrap();
        }
        stack.push(num);
        }
    }
        false
        
    }
}